final-project/static/scripts.js

```diff
@@ -6,23 +6,59 @@
 // Enforce strict js code conventions
 "use strict";
 
-// Render product on screen
-function displayProduct() {
+// Render the product on screen.
+$(function retrieveJSON() {
 
-  // TODO: create parameters for Flask.url_for() method
-  // TODO: use Flask.url_for("product", parameters) to generate JSON url
 
-  // TODO: create variable to store our JSON data
-    // (probably in jqXHR format)
+  // Get the JSON with $.getJSON() & Flask.url_for().
+    // Use Flask.url_for("product", parameters) to generate JSON url.
+  function getProductInfo (parameters, callback) {
+    $.getJSON(Flask.url_for("product"), parameters)
+    // A callback executed upon successful retrieval.
+    .done(function(data, textStatus, jqXHR) {
+      console.log(textStatus)
+      console.log(jqXHR)
+      callback(jqXHR)
```

Comment:
Better to name `callback` just `cb` as a standard.

Reply:
done